src/adapter/src/coord/message_handler.rs

```diff
@@ -164,18 +163,25 @@ impl Coordinator {
     }
 
     #[tracing::instrument(level = "debug", skip_all)]
-    async fn storage_usage_update(&mut self, shard_sizes: BTreeMap<Option<ShardId>, u64>) {
+    async fn storage_usage_update(&mut self, shards_usage: ShardsUsage) {
         // Similar to audit events, use the oracle ts so this is guaranteed to
         // increase. This is intentionally the timestamp of when collection
         // finished, not when it started, so that we don't write data with a
         // timestamp in the past.
         let collection_timestamp: EpochMillis = self.get_local_write_ts().await.timestamp.into();
 
         let mut ops = vec![];
-        for (shard_id, size_bytes) in shard_sizes {
+        for (shard_id, shard_usage) in shards_usage.by_shard {
+            ops.push(catalog::Op::UpdateStorageUsage {
+                shard_id: Some(shard_id.to_string()),
+                size_bytes: shard_usage.referenced_bytes(),
```

Comment:
based on the previous precedent of `shard_sizes.retain` throwing away shards that we no longer consider relevant, I've done the same here with any potential difference between `referenced_bytes` (3a) and `total_bytes` (1, and what we did before this PR). there are a number of other options available though, including:
- put it all under the `shard_id: NULL`/unattributable_bytes entry
- split out `mz_storage_usage_by_shard` into more columns and decide which we use in the `mz_storage_usage` VIEW definition
- something else?

I don't particularly have a horse in this race, does anyone have a strong opinion here?

Reply:
Rami mentioned offline a preference for keeping the schema the same since billing queries have to use it directly. Given that, and the precedent I mentioned, I think I'm gonna assume this is fine and get it merged for next week's release. It's definitely better with this PR and it's hard to imagine how it would be worse. We can always circle back here if someone develops an opinion